### ISSUE TITLE
12$: warn on module with no exports or side effects

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#813]
```javascript
import React from "react";
// the following should warn about missing 'export'
function MyComponent() { return <>hello</>; }
```
